### ISSUE TITLE
[Swift3] Boolean are converted to Strings in Json

### ISSUE THREAD

**Issue by Daxten** [swagger-api/swagger-codegen#5543]
I'm not sure which commit changed this, but the newest client pickles booleans as like this:

"boolprop": "false"

and it should be like this:

"boolprop": false

**Comment by wing328** [comment]
The mapping seems to be correct: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Swift3Codegen.java#L136

so seems like the deserialization logic is not correct.

Would using ObjectMapper fix the issue?

Ref: https://github.com/swagger-api/swagger-codegen/issues/2194

cc @RobBlairInq @jaz-ah @jgavris @hexelon

**Comment by brassel** [comment]
Well, there is this function:

    static func convertBoolToString(_ source: [String: Any]?) -> [String:Any]

This turns Booleans into strings 

    destination[key] = "\(value as! Bool)" as Any?

It has been there before, but in the most recent versions it seems to be called for every called Post request.

    open class func ...PostWithRequestBuilder(...) -> RequestBuilder<Void> {
        var path = "..."
        path = path.replacingOccurrences(...)
        let URLString = SwaggerClientAPI.basePath + path
        let parameters = body.encodeToJSON() as? [String:AnyObject]
 
        // THIS IS NEW!
        let convertedParameters = APIHelper.convertBoolToString(parameters)
 
        let requestBuilder: RequestBuilder<Void>.Type = SwaggerClientAPI.requestBuilderFactory.getBuilder()

        return requestBuilder.init(method: "POST", URLString: URLString, parameters: convertedParameters, isBody: true)
    }

**Comment by wing328** [comment]
@brassel I think convertBoolToString should not be called for your case.

```
    // THIS IS NEW!
    let convertedParameters = APIHelper.convertBoolToString(parameters)
```
We can add a {{^isBoolean}} .. {{/isBoolean}} check in the template to include the code above only if the type is non-boolean. Would that help?

**Comment by brassel** [comment]
Maybe, but it sounds a bit ad hoc, don't you think? I can not quite understand in which cases converting Bools to String is necessary at all. All I know is that just a few commits ago, the method was not called in our generated client and that it does cause an error now. So if you could make that change we could check if it works well again in our case. 
If you would like, you could also describe the intended purpose of this call. Then we could give some better feedback about why it goes wrong in our application.

**Comment by wing328** [comment]
The function convertBoolToString was added by https://github.com/swagger-api/swagger-codegen/pull/2705.

cc @JMDHo to see if he's any feedback on this issue.

**Comment by brassel** [comment]
Our app did still work when the function was added. But recently it is called for our client where it was not before.

**Comment by brassel** [comment]
I have had a look at 

https://github.com/swagger-api/swagger-codegen/issues/2446

which describes *that* the function to convert Booleans to Strings was added as well as

https://github.com/swagger-api/swagger-codegen/issues/2446

which describes that converting Booleans to Strings is not so easy because of an issue with identifying Booleans. This problem and the solution is described here:

http://stackoverflow.com/questions/36185530/how-to-check-dictionary-value-to-be-exactly-a-bool/36186159#36186159

What I can not find, however, is any information *why Boolean values should be converted to Strings in the first place*. Is it possible that Swagger generated server APIs cannot cope with genuine Boolean values and can only work with Strings? Sounds not very probable to me.

**Comment by lightbow** [comment]
convertBoolToString also breaks my app. I don't understand its history or purpose, but if I just remove that function entirely in my sandbox, everything works.

**Comment by brassel** [comment]
> but if I just remove that function entirely in my sandbox, everything works.

Yes, just the same for me. So simply make that function go away, please! :)

**Comment by wing328** [comment]
Please kindly submit a PR so that we can review the fix. Thanks!

**Comment by niebloomj** [comment]
Can confirm I am having the same issue. This conversion breaks my app.

**Comment by gerchicov-bp** [comment]
@brassel 
```destination[key] = "\(value as! Bool)" as Any?```
LOL? If value is not boolean this code throws exception. And of course you don't check that value is boolean

**Comment by brassel** [comment]
@gerchicov-bp, yes quite a lough isn't it? 
So, it was much better before that code was actually called. :o)

I guess you are with me then in saying:
>  So simply make that function go away, please! :)

Cheers
brassel
